src/i18n/languageSwitcher/LanguageSwitcher.tsx

```diff
@@ -26,15 +26,19 @@ function LanguageSwitcher(props: Props) {
   return (
     <ul className={classNames(props.className, styles.list)}>
       {languages.map(lang => (
-        <li key={lang.code} className={styles.item}>
-          <button
-            type="button"
-            lang={lang.code}
-            onClick={() => setLanguage(lang.code)}
-          >
-            {lang.label}
-          </button>
-        </li>
+        <React.Fragment key={lang.code}>
+          {lang.code !== i18n.languages[0] ? (
```

Comment:
Have you considered filtering the unwanted language? It could make for an easier to understand component.

**Also**

Your language code as in the form `language_code` whereas the `i18n.languages` array may be in the format `language_code-locale`.

Reply:
Yep, you were right. If there were no cookies, localStorage or navigation parameters present, `i18next-browser-languagedetector` defaulted language format to `fi-FI`. I added these values to all languages and changed ternary check to filter method.